count.d.ts

```diff
@@ -0,0 +1,146 @@
+interface Window {
+	/**
+	 * GoatCounter JavaScript API
+	 *
+	 * @see https://www.goatcounter.com/help/js
+	 */
+	goatcounter:
+		| (GoatCounter.Settings &
+				GoatCounter.DataParameters &
+				GoatCounter.Methods)
+		| undefined;
+}
+
+declare namespace GoatCounter {
+	/**
+	 * Settings that can be defined via:
+	 *
+	 * - `window.goatcounter` object, declared _before_ loading `count.js`
+	 * - `<script>` tag `data-goatcounter-settings` attribute (will override settings in `window.goatcounter`)
+	 */
+	interface Settings {
+		/** Don’t do anything on page load, for cases where you want to call `count()` manually. Also won’t bind events. */
+		no_onload?: boolean;
+		/** Don’t bind events. */
+		no_events?: boolean;
+		/** Allow requests from local addresses (`localhost`, `192.168.0.0`, etc.) for testing the integration locally. */
+		allow_local?: boolean;
+		/** Allow requests when the page is loaded in a frame or iframe. */
+		allow_frame?: boolean;
+		/** Customize the endpoint for sending pageviews to (overrides the URL in `data-goatcounter`). Only useful if you have `no_onload`. */
+		endpoint?: string;
+	}
+
+	interface DataParameters {
+		/**
+		 * Page path (without domain) or event name.
+		 *
+		 * Default is the value of `<link rel="canonical">` if it exists, or
+		 * `location.pathname + location.search`.
+		 *
+		 * Alternatively, a callback that takes the default value and
+		 * returns a new value. No pageview is sent if the callback returns
+		 * `null`.
+		 *
+		 * @see https://www.goatcounter.com/help/modify
+		 */
```

Comment:
Is using `/* .. */` style comments required? Or will `//` comments also work?

Reply:
Yeah, `//` is fine for single line comments, but this is conventional [JSDoc](https://jsdoc.app/about-getting-started) (which is supported well in most IDEs).